File: Algorithms/Easy/palindrome-index.php
<?php

/**
 * @desc https://www.hackerrank.com/challenges/palindrome-index/
 */

// Complete the palindromeIndex function below.
function palindromeIndex($s) {
    $i = 0;
    $j = strlen($s) - 1;
    for (; $i<=$j; $i++,$j--) {
        if ($s[$i] != $s[$j]) {
            $s1 = substr($s, $i+1, $j-$i);
            $s2 = substr($s, $i, $j-$i);
            if (empty($s1)) {
                return -1;
            }
            $x = 0;
            $y = strlen($s1) - 1;
            for (; $x<=$y; $x++,$y--) {
                if ($s1[$x] != $s1[$y]) {
                    for (; $x<=$y; $x++,$y--) {
                        if ($s2[$x] != $s2[$y]) {
                            return -2;
                        }
                    }
                    return $j;
                }
            }
            return $i;
        }
    }
    return -1;
}

File: Algorithms/Easy/palindrome-index.py
#!/bin/python3

import math
import os
import random
import re
import sys

"""
@url: https://www.hackerrank.com/challenges/palindrome-index/
"""

# Complete the palindromeIndex function below.
def palindromeIndex(s):
    i = 0
    j = len(s)-1
    while i<=j:
        if s[i] is not s[j]:
            left = s[i+1:j+1]
            print(left)
            x = 0
            y = len(left)-1
            while x<=y:
                if left[x] is not left[y]:
                    right = s[i:j]
                    print(right)
                    x = 0
                    y = len(right)-1
                    while x<=y :
                        if right[x] is not right[y]:
                            return -2
                        x += 1
                        y -= 1
                    return j
                x += 1
                y -= 1
            return i
        i += 1
        j -= 1
    return -1